### ISSUE TITLE
Squash rust repo to make it fast to download

### ISSUE THREAD

**Issue by roblabla** [MegatonHammer/megaton-hammer#37]
We should have a rust repo with the whole history squashed to one commit, so that downloading it with Xargo is instantaneous.

**Comment by roblabla** [comment]
I think (but would need to check) that xargo just uses cargo? So it might be cargo that needs to do that. If it doesn't already.

**Comment by gerd2002** [comment]
last time I checked this wasn't possible.

**Comment by kitlith** [comment]
right, there's an issue on cargo: https://github.com/rust-lang/cargo/issues/1171

which is blocked on an issue in libgit2....

**Comment by roblabla** [comment]
for reference (took me a while to find it), the libgit2 issue: https://github.com/libgit2/libgit2/issues/3058
